employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/common/EventManager.java

```diff
@@ -118,31 +118,33 @@ public void handleEvent(T value) {
-        public static Event<Pagination> SHIFT_ROSTER_PAGINATION = new Event<>("ShiftRosterPagination", Pagination.class);
-        public static Event<LocalDateRange> SHIFT_ROSTER_DATE_RANGE = new Event<>("ShiftRosterDateRange", LocalDateRange.class);
+        public static final Event<ShiftRosterView> SHIFT_ROSTER_UPDATE = new Event<>("ShiftRosterUpdate", ShiftRosterView.class);
+        public static final Event<Void> SHIFT_ROSTER_INVALIDATE = new Event<>("ShiftRosterInvalidate", Void.class);
+        public static final Event<Pagination> SHIFT_ROSTER_PAGINATION = new Event<>("ShiftRosterPagination", Pagination.class);
+        public static final Event<LocalDateRange> SHIFT_ROSTER_DATE_RANGE = new Event<>("ShiftRosterDateRange", LocalDateRange.class);
 
         // Employee Roster Events
-        public static Event<AvailabilityRosterView> AVAILABILITY_ROSTER_UPDATE = new Event<>("AvailabilityRosterUpdate", AvailabilityRosterView.class);
-        public static Event<Void> AVAILABILITY_ROSTER_INVALIDATE = new Event<>("AvailabilityRosterInvalidate", Void.class);
-        public static Event<Pagination> AVAILABILITY_ROSTER_PAGINATION = new Event<>("AvailabilityRosterPagination", Pagination.class);
-        public static Event<LocalDateRange> AVAILABILITY_ROSTER_DATE_RANGE = new Event<>("AvailabilityRosterDateRange", LocalDateRange.class);
+        public static final Event<AvailabilityRosterView> AVAILABILITY_ROSTER_UPDATE = new Event<>("AvailabilityRosterUpdate", AvailabilityRosterView.class);
+        public static final Event<Void> AVAILABILITY_ROSTER_INVALIDATE = new Event<>("AvailabilityRosterInvalidate", Void.class);
+        public static final Event<Pagination> AVAILABILITY_ROSTER_PAGINATION = new Event<>("AvailabilityRosterPagination", Pagination.class);
+        public static final Event<LocalDateRange> AVAILABILITY_ROSTER_DATE_RANGE = new Event<>("AvailabilityRosterDateRange", LocalDateRange.class);
 
         // Rotation Events
-        public static Event<Void> ROTATION_SAVE = new Event<>("RotationSave", Void.class);
-        public static Event<Void> ROTATION_INVALIDATE = new Event<>("RotationInvalidate", Void.class);
+        public static final Event<Void> ROTATION_SAVE = new Event<>("RotationSave", Void.class);
+        public static final Event<Void> ROTATION_INVALIDATE = new Event<>("RotationInvalidate", Void.class);
 
         // Invalidation Events
         @SuppressWarnings("rawtypes")
-        public static Event<Class> DATA_INVALIDATION = new Event<>("DataInvalidation", Class.class);
+        public static final Event<Class> DATA_INVALIDATION = new Event<>("DataInvalidation", Class.class);
         @SuppressWarnings("rawtypes")
-        public static Event<Map> SKILL_MAP_INVALIDATION = new Event<>("SkillMapInvalidation", Map.class);
+        public static final Event<Map> SKILL_MAP_INVALIDATION = new Event<>("SkillMapInvalidation", Map.class);
+        @SuppressWarnings("rawtypes")
```

Comment:
In a separate PR, we'll want to remove these "raw types" ignores, or comment why `Event<Map<Object, Object>>` isn't possible, because this half-use of generics leads to full problems.

Reply:
Unfortunately this will require new classes (something like `SkillMap extends HashMap<String, Skill>`) and new events for data invalidation but I think it's the correct way to do it. Having a dedicated event for each data type might be some extra code but on the other hand it will allow to simplify this:
```java
eventManager.subscribeToEventForever(Event.DATA_INVALIDATION, clazz -> {
    if (clazz.equals(Employee.class)) {
        updateRowCount();
    }
});
```
to this:
```java
eventManager.subscribeToEventForever(Event.EMPLOYEE_INVALIDATION, aVoid -> updateRowCount());
```

I can do that.